store/store.go

```diff
@@ -58,6 +59,19 @@ func (s *DefaultStore) Height() uint64 {
 	return atomic.LoadUint64(&s.height)
 }
 
+// SetBase sets the height saved in the Store of the earliest block
+func (s *DefaultStore) SetBase(height uint64) {
+	baseHeight := atomic.LoadUint64(&s.baseHeight)
+	if height > baseHeight {
+		_ = atomic.CompareAndSwapUint64(&s.baseHeight, baseHeight, height)
+	}
+}
+
+// Base returns height of the earliest block saved in the Store.
+func (s *DefaultStore) Base() uint64 {
```

Comment:
Not sure I see where base is initialized after a restart? Maybe I missed that. 
So for example if base is 100, and I restart the process, it seems as if it will return to be 0.
For example the height is saved in the state and initialized in store.go line 210 under LoadState() function.

Reply:
u right. I've tried to avoid changes to the state, and the `0-100` blocks in your example will be skipped when pruning for the first time after restert.
```
for h := base; h < height; h++ {
		hash, err := s.loadHashFromIndex(h)
		if err != nil {
			continue
		}
```

U think it should be added to the state?

Reply:
Actually the major problem I'm seeing here is that if base=0 after restart, and `retainHeight` returned from the app in the [pruning.go](https://github.com/dymensionxyz/dymint/pull/328/commits/2509b69582d1b51edd7b7a2f8f43cda140df5735#diff-812ea47bd8817fdc1e28a12177c62dfb338c3679ae9e0a132532ac4e09d83db8) is bigger, then in line 6 seems like it will never actually call `store.pruneBlocks` and as a result will never update the base

Reply:
you're right. my miss here. 

but still it seems like if the `retainHeight` is in the order of millions and we restart the rollapp and base is 0, so  this will be iterated for a million times without a reason:

```
for h := base; h < height; h++ {
		hash, err := s.loadHashFromIndex(h)
		if err != nil {
			continue
		}
```

Reply:
U correct, but I'm not sure it's worth the effort now to keep the Base in the persistent store.

I would leave it as enhancement, + adding pruning in intervals which currently not implemented
(i.e a pruning will happen after each Commit)

If u think it should be handled in this PR, let me know
Otherwise, approve and I'll open a separate issue

Reply:
I think it's relatively a small effort to add the base to the persisted state and IMO it's very likely that rollups will restart after having a huge height which will in turn cause millions of redundant I/O calls. 

IMO the feature is not complete as I don't see persisting the base as an enhancement but as a bug if it doesn't exist (as base is expected to be something else upon restart). 

Re pruning intervals, I agree it's an enhancement.